src/components/member-page/donation/download-page.js

```diff
@@ -0,0 +1,377 @@
+
+const RecordBoxHeader = styled.div`
+  width: 100%;
+  height: 32px;
+  gap: 16px;
+  border-bottom: 1px solid ${colorGrayscale.gray200};
+  display: flex;
+  align-items: center;
+  .date {
+    width: 20%;
+  }
+  .order-number {
+    width: 40%;
+  }
+  .amount {
+    width: 20%;
+  }
+  .status {
+    width: 20%;
+  }
+`
+
+const RecordBoxContent = styled.div`
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+  @media print {
+    * {
+      break-inside: avoid;
+    }
+  }
+`
+
+const RecordBoxContentRow = styled.div`
+  width: 100%;
+  height: 32px;
+  gap: 16px;
+  display: flex;
+  align-items: center;
+  &:nth-child(even) {
+    background-color: ${colorOpacity['black_0.05']};
+  }
+  .date {
+    width: 20%;
+  }
+  .order-number {
+    width: 40%;
+  }
+  .amount {
+    width: 20%;
+  }
+  .status {
+    width: 20%;
+  }
+`
+
+const DotGroup = styled.div`
+  display: flex;
+  gap: 4px;
+`
+
+const Dot = styled.div`
+  width: 4px;
+  height: 4px;
+  border-radius: 50%;
+  background-color: ${colorGrayscale.black};
+`
+
+const Loading = styled.div``
+
+const LoadingMask = FetchingWrapper(Loading)
+
+const DotGroups = () =>
+  Array.from({ length: 3 }, (_, i) => (
+    <DotGroup key={i}>
+      <Dot />
+      <Dot />
+      <Dot />
+      <Dot />
+    </DotGroup>
+  ))
+
+const { reduxStateFields } = twreporterRedux
+
+const statusDictionary = {
+  paying: '進行中',
+  paid: '已完成',
+  fail: '扣款失敗',
+  refunded: '已退款',
+}
+
+const DonwloadPage = () => {
+  const { orderNumber } = useParams()
+  const store = useStore()
+  const [state] = store
+  const [history, setHistory] = useState()
+  const [receiptHeader, setReceiptHeader] = useState('')
+  const [receiptAddress, setReceiptAddress] = useState('')
+  const [receiptPayMethod, setReceiptPayMethod] = useState('')
+  const [cardLastFour, setCardLastFour] = useState('')
+  const [cardType, setCardType] = useState('')
+  const [isLoading, setIsLoading] = useState(true)
+  const jwt = _.get(state, [reduxStateFields.auth, 'accessToken'])
+  const apiOrigin = _.get(state, [reduxStateFields.origins, 'api'])
+
+  const getReceiptData = async () => {
+    const { data: orderResponseData } = await axios.get(
+      `${apiOrigin}/v1/periodic-donations/orders/${orderNumber}`,
+      {
+        timeout: 8000,
+        headers: {
+          Authorization: `Bearer ${jwt}`,
+        },
+        withCredentials: true,
+      }
+    )
+    const {
+      card_info: cardInfo,
+      receipt,
+      pay_method: payMethod,
+    } = orderResponseData.data
+    const address = `${receipt.address_state || ''}${receipt.address_city ||
+      ''}${receipt.address_detail || ''}`
+    setReceiptHeader(receipt.header)
+    setReceiptAddress(address)
+    setReceiptPayMethod(payMethod)
+    setCardLastFour(cardInfo.last_four)
+    setCardType(cardInfo.type)
+  }
+  const getPaymentHistory = async () => {
+    // set limit to -1 to get all records
+    const { data: historyResponseData } = await axios.get(
```

Comment:
same as [this comment](https://github.com/twreporter/twreporter-react/pull/2837/files#r1592915147)

Reply:
fixed